### PR TITLE
[BREAKINGCHANGE] Upgrade to React 18.2.0

### PR DESCRIPTION
**Pull request by Gladorme**
# Description

Upgrade all package to React 18

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky. See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests) and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/prometheus-plugin/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "immer": "^9.0.15"
   },
   "peerDependencies": {
-    "react": "^17.0.2 || ^18.0.0"
```
Comment (Nexucis) [inline ui/prometheus-plugin/package.json:43]: do we have a reason to remove the support of react v17 ?
I don't see any changes in this PR that required this drop, am I missing something ?

Comment (Gladorme) [reply ui/prometheus-plugin/package.json:43]: If we want to support newer React features #1353, React 17 support will need to drop

Comment (Nexucis) [reply ui/prometheus-plugin/package.json:43]: yeah I agree but that's not the case for the moment excepting if you are willing to do it :)

Comment (Gladorme) [reply ui/prometheus-plugin/package.json:43]: Updated ❤️

**Review by Nexucis — COMMENTED** [review]
sounds good to me. @jgbernalp is ok on RedHat side to drop the support of React 17 ?

**Comment by jgbernalp** [comment]
@Gladorme Unfortunately I think it might break some components for us. Is there a strong reason why we should change the `useId` hook already provided by perses components package?

**Comment by Gladorme** [comment]
@jgbernalp No strong reasons, I saw we had some warnings about deps vulnerability and I wanted to update of our deps. I know we had an topic about React 17 -> React 18, so I started by this 😄 
Let's put in standby for now

**Comment by Nexucis** [comment]
@jgbernalp is it still not ok to drop the support of react v17 ?

**Comment by jgbernalp** [comment]
@Nexucis unfortunately yes. We still depend on react 17 for dynamic plugins on the OpenShift console, this is to add support for Perses dashboards to web consoles for older OpenShift versions. But we can make an assessment of the impact once we have the console plugin ready for distributed tracing and revisit this PR.

**Comment by Nexucis** [comment]
let's close this PR.
When we will be able to make the upgrade there will be more changes anyway. So this PR cannot be merged just like that.

**Pull request closed without merging**